### PR TITLE
[F] add daylight widget

### PR DESCRIPTION
**Pull request by alexgoff**
Daylight widget to show the night time, twilight, and daylight hours at the Rubin Observatory. Includes accessible interpretations of daylight data.

### CODE REVIEW

File: components/dynamic/SummitData/Astroweather/Current/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { useState } from "react";
+import SunCalc from "suncalc";
+import { useTranslation } from "react-i18next";
+import { altitude, lat, long, timezone } from "@/lib/observatory";
+import { timezoneOffset } from "@/helpers";
+import WidgetSection from "@/components/layout/WidgetSection";
+import Daylight from "@/components/widgets/CurrentData/patterns/Daylight";
+
+const CurrentAstroweather = () => {
```
Comment (alexgoff) [inline components/dynamic/SummitData/Astroweather/Current/index.js:9]: section for current astro weather which right now is just daylight hours

File: components/widgets/CurrentData/patterns/Daylight/index.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+              "summit_dashboard.sections.astro.daylight.observing",
+              {
+                from: formatTime(night, language),
+                to: formatTime(midnight, language),
+              }
+            )}
+          />
+        </g>
+        <g>
+          <Styled.Now
```
Comment (alexgoff) [inline components/widgets/CurrentData/patterns/Daylight/index.js:187]: the "now" indicator will read out the current time and if it is dark enough to observe

File: helpers/index.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-export const capitalize = (string) => {
+export const capitalize = (string, locale = "en-US") => {
   if (typeof string !== "string") return "";
-  return string.charAt(0).toUpperCase() + string.slice(1);
+  return string.charAt(0).toLocaleUpperCase(locale) + string.slice(1);
```
Comment (alexgoff) [inline helpers/index.js:3]: use optional locale setting with locale uppercase

File: lib/observatory.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
 export const lat = -30.244639;
 export const long = -70.749417;
 export const timezone = "America/Santiago";
+export const altitude = 2663;
```
Comment (alexgoff) [inline lib/observatory.js:4]: altitude lets us get the correct sunrise/set times

**Review by blnkt — CHANGES_REQUESTED** [review]
Amazing work.  The accessibility thoughtfulness is 👍 .   A couple small stylistic things:

1. Not technically a part of this PR, but I didn't notice this before: the crescent paths on the moon icon are missing a black fill.  I think it's always black for every phase (?) so you can probs inline the fill on the path element itself, rather than add `fill: #000` to all of the phase pathProps objects? 
Screen Shot 2023-06-28 at 10 09 37 AM

2. Also in the moon phase widget, the lower phase label gets a little smooshed on smaller screens.  Match text style of the upper label?
Screen Shot 2023-06-28 at 9 58 08 AM

3. The DayLight Legend overflows out of its panel on smaller screens
Screen Shot 2023-06-28 at 9 39 38 AM

File: components/atomic/WidgetBackground/styles.js [hunk 5]
```diff
@@ -8,13 +9,14 @@ const variants = {
 export const WidgetBackground = styled.div`
   --widget-background-color: ${({ $variant = "primary" }) =>
     variants[$variant]};
+  --widget-padding: ${fluidScale("20px", "10px")};
```
Comment (alexgoff) [inline components/atomic/WidgetBackground/styles.js:12]: scale widget padding based on screen size

File: components/charts/ChartBase/styles.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import styled from "styled-components";
+
+export const SVG = styled.svg`
+  aspect-ratio: var(--aspect-ratio);
+  min-width: var(--min-width);
+  width: 100%;
+  max-height: 100%;
+  margin-inline: auto;
```
Comment (alexgoff) [inline components/charts/ChartBase/styles.js:8]: use some old tricks to get centering + overflow when applicable + no vertical overflow

File: components/layout/WidgetPreview/styles.js [hunk 7]
```diff
@@ -1,7 +1,20 @@
...
+  large: 1,
+  full: 1,
+};
+const tabletColumns = {
+  small: 1,
+  medium: 1,
+  large: 2,
+  full: 3,
+};
+const desktopColumns = {
```
Comment (alexgoff) [inline components/layout/WidgetPreview/styles.js:17]: change the number of columns in each widget preview at different screen sizes

**Pull request merged**